File: codes/algorithm/select_sort.cpp
#include <iostream>
#include <vector>
using namespace std;


int select_sort(vector<int> &arr){
    for(int i = 0; i < arr.size()-1; i++){
        int min_idx = i;
        for (int j = i; j < arr.size(); j++){
            if(arr[j] < arr[min_idx]) min_idx = j;
        }
        swap(arr[i], arr[min_idx]);
    }
    // return arr;
    for(auto &e : arr)
        cout << e << " ";
    cout << endl;
    return 0;
}

int main(){
    vector<int> arr = {4,1,2,3,6,5};
    select_sort(arr);
}

File: codes/algorithm/select_sort.py
def select_sort(target):
    for i in range(len(target)-1):
        min_idx = i
        for j in range(i, len(target)):
            if target[j] < target[min_idx]: # 부등호 반대하면 내림차순
                min_idx = j
        target[i], target[min_idx] = target[min_idx], target[i]
    return target

print(select_sort([5,2,3,4,1]))
        
